### PR TITLE
Upgrade @primer/stylelint-config to latest version 13.0.1

### PR DESCRIPTION
**Pull request by jonrohan**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

#### List the issues that this change affects.

Closes # (type the GitHub issue number after #)

#### Risk Assessment
  

- [ ] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **Fixes axe scan violation** - This change fixes an existing [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation.
- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.
- **New axe violation** - This change introduces a new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation. Please describe why the violation cannot be resolved below.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/alpha/action_list.pcss [hunk 1]
```diff
@@ -492,7 +502,7 @@ nav-list {
 
     & .ActionListItem-multiSelectIconRect {
       fill: var(--bgColor-default);
-      border: var(--borderWidth-thin, 1px) solid var(--control-borderColor-rest);
+      border: var(--borderWidth-thin, var(--borderWidth-thin)) solid var(--control-borderColor-rest);
```
Comment (langermank) [inline app/components/primer/alpha/action_list.pcss:505]:
```suggestion
      border: var(--borderWidth-thin) solid var(--control-borderColor-rest);
```

File: app/components/primer/alpha/text_field.pcss [hunk 2]
```diff
@@ -33,8 +39,9 @@
 .FormControl-inlineValidation {
   display: flex;
   font-size: var(--text-caption-size);
-  line-height: var(--text-caption-lineHeight);
   font-weight: var(--base-text-weight-semibold);
+  line-height: var(--text-caption-lineHeight);
+  /* stylelint-disable-next-line primer/colors */
   color: var(--control-borderColor-danger);
   fill: var(--control-borderColor-danger);
```
Comment (langermank) [inline app/components/primer/alpha/text_field.pcss:46]:
```suggestion
  color: var(--control-danger-fgColor-rest);
  fill: var(--control-danger-fgColor-rest);
```

File: app/components/primer/alpha/text_field.pcss [hunk 3]
```diff
@@ -47,6 +54,7 @@
 }
 
 .FormControl-inlineValidation--success {
+  /* stylelint-disable-next-line primer/colors */
   color: var(--control-borderColor-success);
   fill: var(--control-borderColor-success);
```
Comment (langermank) [inline app/components/primer/alpha/text_field.pcss:59]:
```suggestion
  color: var(--fgColor-success);
  fill: var(--fgColor-success);
```

File: app/components/primer/alpha/text_field.pcss [hunk 4]
```diff
@@ -154,7 +162,7 @@
   transition: 80ms cubic-bezier(0.33, 1, 0.68, 1);
   transition-property: color, background-color, box-shadow, border-color;
   padding-inline: var(--control-medium-paddingInline-condensed);
-  padding-block: calc(var(--control-medium-paddingBlock, 6px) - var(--borderWidth-thin, 1px));
+  padding-block: calc(var(--control-medium-paddingBlock, var(--base-size-6)) - var(--borderWidth-thin, 1px));
```
Comment (langermank) [inline app/components/primer/alpha/text_field.pcss:165]:
```suggestion
  padding-block: calc(var(--control-medium-paddingBlock) - var(--borderWidth-thin));
```

File: app/components/primer/alpha/toggle_switch.pcss [hunk 5]
```diff
@@ -173,14 +175,15 @@
...
   color: var(--fgColor-default);
   text-align: right;
 }
 
 .ToggleSwitch-statusIcon {
-  width: var(--base-size-16);
   display: flex;
+  width: var(--base-size-16);
+  /* stylelint-disable-next-line primer/spacing */
   margin-top: 0.063rem;
```
Comment (langermank) [inline app/components/primer/alpha/toggle_switch.pcss:187]: holy cow

File: app/components/primer/beta/border_box.pcss [hunk 6]
```diff
@@ -59,7 +62,7 @@
   & .Box-btn-octicon {
     &.btn-octicon {
       padding: var(--stack-padding-spacious);
-      margin: calc(var(--stack-gap-spacious) * -1) calc(var(--stack-gap-spacious) * -1);
+      margin: calc(var(--stack-gap-spacious) * -1);
```
Comment (langermank) [inline app/components/primer/beta/border_box.pcss:65]: this change looks sus

File: app/components/primer/beta/border_box.pcss [hunk 7]
```diff
@@ -277,8 +281,8 @@
 .Box-btn-octicon {
   /* Increase specificity when btn-octicon is used because comes after .Box in the cascade */
   &.btn-octicon {
-    padding: var(--control-medium-paddingInline-spacious) var(--control-medium-paddingInline-spacious);
-    margin: calc(var(--controlStack-small-gap-spacious) * -1) calc(var(--controlStack-small-gap-spacious) * -1);
-    line-height: 1.5; /* override btn-octicon line-height */
+    padding: var(--control-medium-paddingInline-spacious);
+    margin: calc(var(--controlStack-small-gap-spacious) * -1);
```
Comment (langermank) [inline app/components/primer/beta/border_box.pcss:285]: Same with these, but not sure?

File: app/components/primer/beta/border_box.pcss (hunk 6 above)
Comment (jonrohan) [reply app/components/primer/beta/border_box.pcss:65]: I ran into a rule about shorthand duplication, since the 2 values for top and sides was the same

**Review by langermank — APPROVED** [review]

**Pull request merged**